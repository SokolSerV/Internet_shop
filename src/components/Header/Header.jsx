import { useSelector } from "react-redux";

import Navigation from "../Navigation";

import { ReactComponent as Logo } from "./icons/logo.svg";
import { ReactComponent as FavouriteIcon } from "./icons/favourite.svg";
import { ReactComponent as BasketIcon } from "./icons/basket.svg";

import "./Header.scss";

export default function Header() {
    const favouriteCounter = useSelector(
        (state) => state.favourite.favouriteList.length
    );

    const basketCounter = useSelector(
        (state) => state.basket.basketList.length
    );

    return (
        <header className="header">
            <div className="header_wrapper">
                <span className="header_logo">
                    <Logo />
                </span>
                <p className="header_title">Сторінка інтернет-магазину</p>
                <Navigation />
                <div className="header_icons">
                    <span className="favorites">
                        <FavouriteIcon />
                        <span className="counter">{favouriteCounter}</span>
                    </span>
                    <span className="basket">
                        <BasketIcon />
                        <span className="counter">{basketCounter}</span>
                    </span>
                </div>
            </div>
        </header>
    );
}
