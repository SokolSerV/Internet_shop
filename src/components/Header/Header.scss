.header {
    background-color: lightseagreen;
    padding: 5px 50px;
    position: sticky;
    top: 0;
}

.header_wrapper {
    display: flex;
    align-items: center;
}

.header_title {
    margin: 0;
    padding-left: 10px;
    font-size: 20px;
    line-height: 20px;
    font-weight: 700;
    color: whitesmoke;
    text-transform: uppercase;
}

.header_icons {
    display: flex;
    gap: 15px;
    padding: 5px 10px;
    background-color: grey;
    border-radius: 20px;
}

.favorites,
.basket {
    display: flex;
}

// .favorites_svg,
// .basket_svg {
//     cursor: pointer;
// }

.counter {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 15px;
    height: 15px;
    background-color: lightyellow;
    border-radius: 50%;
    font-size: 0.75em;
}
