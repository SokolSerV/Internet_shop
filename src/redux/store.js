import { configureStore } from "@reduxjs/toolkit";
import goodsSlice from "./slices/goodsSlice";
import favouriteSlice from "./slices/favouriteSlice";
import basketSlice from "./slices/basketSlice";
import modalSlice from "./slices/modalSlice";

const store = configureStore({
    reducer: {
        goods: goodsSlice,
        favourite: favouriteSlice,
        basket: basketSlice,
        modal: modalSlice,
    },
});

export default store;
