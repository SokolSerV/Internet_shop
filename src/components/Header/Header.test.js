import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../redux/store";

import Header from "./Header";

const Component = () => {
    return (
        <Provider store={store}>
            <Header />
        </Provider>
    );
};

jest.mock("../Navigation", () => () => <div>Navigation</div>);

describe("Header component Smoke test (Snapshot test)", () => {
    test("should Header render", () => {
        const { asFragment } = render(<Component />);

        expect(asFragment()).toMatchSnapshot();
    });
});
