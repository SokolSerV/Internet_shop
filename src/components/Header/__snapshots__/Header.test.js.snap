// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Smoke test (Snapshot test) should Header render 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header_wrapper"
    >
      <span
        class="header_logo"
      >
        <svg>
          logo.svg
        </svg>
      </span>
      <p
        class="header_title"
      >
        Сторінка інтернет-магазину
      </p>
      <div>
        Navigation
      </div>
      <div
        class="header_icons"
      >
        <span
          class="favorites"
        >
          <svg>
            favourite.svg
          </svg>
          <span
            class="counter"
          >
            0
          </span>
        </span>
        <span
          class="basket"
        >
          <svg>
            basket.svg
          </svg>
          <span
            class="counter"
          >
            0
          </span>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
